src/main.rs

```diff
@@ -1,205 +1,342 @@
+    curl.http_09_allowed(true)?;
 
-        let mut headers = List::new();
-        let mut client_header = "X-SwiftNav-Client-Id: ".to_string();
-        client_header.push_str(&opt.client_id);
+    if opt.verbose {
+        curl.verbose(true)?;
+    }
 
-        headers.append("Transfer-Encoding:")?;
-        headers.append("Ntrip-Version: Ntrip/2.0")?;
-        headers.append(&client_header)?;
+    if let Some(username) = &opt.username {
+        curl.username(username)?;
+    }
 
-        curl.http_headers(headers)?;
-        curl.useragent("NTRIP ntrip-client/1.0")?;
-        curl.url(&opt.url)?;
-        curl.progress(true)?;
-        curl.put(true)?;
-        curl.custom_request("GET")?;
-        curl.http_version(HttpVersion::Any)?;
-        curl.http_09_allowed(true)?;
+    if let Some(password) = &opt.password {
+        curl.password(password)?;
+    }
 
-        if opt.verbose {
-            curl.verbose(true)?;
+    let (tx, rx) = flume::bounded::<Vec<u8>>(1);
+    let transfer = Rc::new(RefCell::new(curl.transfer()));
+
+    transfer.borrow_mut().progress_function({
+        let rx = &rx;
+        let transfer = Rc::clone(&transfer);
+        move |_dltot, _dlnow, _ultot, _ulnow| {
+            if !rx.is_empty() {
+                if let Err(e) = transfer.borrow().unpause_read() {
+                    eprintln!("unpause error: {e}");
+                    return false;
+                }
+            }
+            true
         }
+    })?;
 
-        if let Some(username) = &opt.username {
-            curl.username(username)?;
+    transfer.borrow_mut().write_function(|data| {
+        if let Err(e) = io::stdout().write_all(data) {
+            eprintln!("write error: {e}");
+            return Ok(0);
         }
+        Ok(data.len())
+    })?;
 
-        if let Some(password) = &opt.password {
-            curl.password(password)?;
+    transfer.borrow_mut().read_function(|mut data: &mut [u8]| {
+        let Ok(bytes) = rx.try_recv() else {
+            return Err(ReadError::Pause);
+        };
+        if let Err(e) = data.write_all(&bytes) {
+            eprintln!("read error: {e}");
+            return Err(ReadError::Abort);
         }
+        Ok(bytes.len())
+    })?;
 
-        curl.write_function(|buf| Ok(io::stdout().write_all(buf).map_or(0, |_| buf.len())))?;
-
-        curl.progress_function(|_dltot, _dlnow, _ultot, _ulnow| {
-            let now = SystemTime::now();
-            let elapsed = LAST.with(|last| {
-                let dur = now.duration_since(*last.borrow());
-                dur.unwrap_or_else(|_| Duration::from_secs(0)).as_secs()
-            });
-            if elapsed > 10 {
-                CURL.with(|curl| curl.borrow().unpause_read().unwrap());
+    let handle = thread::spawn(move || {
+        let commands = get_commands(opt.clone())?;
+        for cmd in commands {
+            if cmd.after > 0 {
+                thread::sleep(Duration::from_secs(cmd.after));
             }
```

Comment:
Sorry to nitpick but if someone sets after to be zero in the file this conditional still effectively spams the service via curl. Maybe this is desired behavior but in that case i would just remove the conditional. That or
```suggestion
            if cmd.after > 0 {
                thread::sleep(Duration::from_secs(cmd.after));
            } else {
                thread::sleep(Duration::from_secs(10));
            }
```

Reply:
Ahh that's intentional, we do want the ability to test a gga immediately followed by an on demand request

Reply:
Waiting for a second would also probably be fine